src/main/java/inventory/Dispense.java

```diff
@@ -8,17 +8,46 @@
  */
 
 public class Dispense extends Medicine {
+    public static final String ID = "ID";
+    public static final String NAME = "NAME";
+    public static final String QUANTITY = "QUANTITY";
+    public static final String CUSTOMERS_NRIC = "CUSTOMER'S NRIC";
+    public static final String DATE = "DATE";
+    public static final String STAFF_NAME = "STAFF NAME";
+
+    public static final String[] COLUMNS = {ID, NAME, QUANTITY, CUSTOMERS_NRIC, DATE, STAFF_NAME};
+
+    private static int dispenseCount = 0;
+    protected int dispenseId;
     protected String customerNric;
     protected Date date;
     protected String staffName;
 
-    public Dispense(String medicineName, int quantity, String customerName, Date date, String staffName) {
+    public Dispense(String medicineName, int quantity, String customerNric, Date date, String staffName) {
         super(medicineName, quantity);
-        this.customerNric = customerName;
+        dispenseCount++;
+        this.dispenseId = dispenseCount;
+        this.customerNric = customerNric;
         this.date = date;
         this.staffName = staffName;
     }
 
+    public static int getDispenseCount() {
+        return dispenseCount;
+    }
+
+    public static void setDispenseCount(int dispenseCount) {
+        Dispense.dispenseCount = dispenseCount;
+    }
+
```

Comment:
Perhaps can remove setDispenseCount and setDispenseId methods

Reply:
We might need this when loading data from the file so that we can ensure the stock count is accurate. Will discuss in the upcoming meeting.